src/Windows/Avalonia.Win32/OleDataObjectHelper.cs

```diff
@@ -37,15 +42,78 @@ public static FORMATETC ToFormatEtc(this DataFormat format)
             return null;
 
         var medium = new STGMEDIUM();
-        if (oleDataObject.GetData(&formatEtc, &medium) != (uint)HRESULT.S_OK)
-            return null;
+        var result = oleDataObject.GetData(&formatEtc, &medium);
+        if (result != (uint)HRESULT.S_OK)
+        {
+            if (result == 0x80040069) // DV_E_TYMED
+            {
+                formatEtc.tymed = TYMED.TYMED_GDI;
+
+                if (oleDataObject.GetData(&formatEtc, &medium) != (uint)HRESULT.S_OK)
+                {
+                    return null;
+                }
+            }
+            else
+                return null;
+        }
 
         try
         {
-            if (medium.tymed == TYMED.TYMED_HGLOBAL && medium.unionmember != IntPtr.Zero)
+            if (medium.unionmember != IntPtr.Zero)
             {
-                var hGlobal = medium.unionmember;
-                return ReadDataFromHGlobal(format, hGlobal);
+                if (medium.tymed == TYMED.TYMED_HGLOBAL)
+                {
+                    var hGlobal = medium.unionmember;
+                    return ReadDataFromHGlobal(format, hGlobal, formatEtc);
+                }
+                else if (medium.tymed == TYMED.TYMED_GDI)
+                {
+                    var bitmapHandle = medium.unionmember;
+                    var bitmap = new BITMAP();
+                    unsafe
+                    {
+                        var pBitmap = &bitmap;
+                        GetObject(bitmapHandle, Marshal.SizeOf(bitmap), (IntPtr)pBitmap);
+
+                        var bitmapInfoHeader = new BITMAPINFOHEADER()
+                        {
+                            biWidth = bitmap.bmWidth,
+                            biHeight = bitmap.bmHeight,
+                            biPlanes = bitmap.bmPlanes,
+                            biBitCount = 32,
+                            biCompression = 0,
+                            biSizeImage = (uint)(bitmap.bmWidth * 4 * Math.Abs(bitmap.bmHeight))
+                        };
+
+                        bitmapInfoHeader.Init();
+
+                        var destHdc = UnmanagedMethods.GetDC(IntPtr.Zero);
```

Comment:
A lot of the following native calls can fail. We should abort if anything goes wrong.